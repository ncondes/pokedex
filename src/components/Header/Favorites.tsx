'use client'

import Link from 'next/link'
import { FC } from 'react'
import { usePathname } from 'next/navigation'

export const Favorites: FC = () => {
   const pathname = usePathname()

   return (
      <Link aria-label="favorites link" href="/favorites">
         {pathname === '/favorites' ? (
            <i className="fa-solid fa-heart fa-lg"></i>
         ) : (
            <i className="fa-regular fa-heart fa-lg"></i>
         )}
      </Link>
   )
}
