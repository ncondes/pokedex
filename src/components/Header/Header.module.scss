@use '@/scss/colors' as *;

.header {
   height: 3.5rem;
   background-color: $blue-primary-color;
   padding: 0 1.5rem;

   ul {
      display: flex;
      justify-content: space-between;
      align-items: center;
      height: 100%;
      list-style: none;
   }
}

.home_link {
   display: flex;
   align-items: center;
   height: 100%;

   span {
      margin-left: 0.5rem;
      color: $yellow-secondary-color;
      font-size: 1.5rem;
      font-weight: 500;
      text-shadow: 1px 1px 2px rgba(0, 0, 0, 0.2);
   }
}

.theme_and_favorites_links {
   display: flex;
   justify-content: center;
   align-items: center;

   li {
      margin-left: 1rem;
   }

   button {
      background-color: transparent;
   }

   i {
      color: $white-cream-color;
   }
}
