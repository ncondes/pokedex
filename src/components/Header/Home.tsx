import Image from 'next/image'
import Link from 'next/link'
import { FC } from 'react'
import styles from './Header.module.scss'

export const Home: FC = () => {
   return (
      <Link href="/" className={styles.home_link}>
         <div>
            <Image
               src={'/images/pokeball-icon-color.png'}
               alt="pokeball icon"
               width={25}
               height={25}
            />
         </div>
         <span>Pokedex</span>
      </Link>
   )
}
