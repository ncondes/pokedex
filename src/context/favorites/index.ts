export { FavoritesContext } from './FavoritesContext'
export { FavoritesProvider } from './FavoritesProvider'
