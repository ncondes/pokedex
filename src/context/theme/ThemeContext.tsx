'use client'

import { createContext } from 'react'

interface Props {
   theme: 'light' | 'dark'
   toggleTheme: () => void
   useTheme: (className: string) => string
}

export const ThemeContext = createContext({} as Props)
