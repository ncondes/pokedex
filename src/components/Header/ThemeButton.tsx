'use client'

import { ThemeContext } from '@/context/theme'
import { FC, useContext } from 'react'

export const ThemeButton: FC = () => {
   const { theme, toggleTheme } = useContext(ThemeContext)

   const handleClick = () => toggleTheme()

   return (
      <button onClick={handleClick}>
         <i
            className={`fa-regular ${
               theme === 'light' ? 'fa-moon' : 'fa-sun'
            } fa-xl`}
         ></i>
      </button>
   )
}
