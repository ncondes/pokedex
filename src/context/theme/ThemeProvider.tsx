'use client'

import { FC, PropsWithChildren, useState } from 'react'
import { ThemeContext } from './ThemeContext'
import { getLocalTheme, setLocalTheme } from '@/utils'

const INITIAL_THEME = getLocalTheme() as 'light' | 'dark'

export const ThemeProvider: FC<PropsWithChildren> = ({ children }) => {
   const [theme, setTheme] = useState<'light' | 'dark'>(INITIAL_THEME)

   const toggleTheme = () => {
      const t = theme === 'light' ? 'dark' : 'light'
      setTheme(t)
      setLocalTheme(t)
   }

   const useTheme = (className: string): string => `${theme}-${className}`

   return (
      <ThemeContext.Provider value={{ theme, toggleTheme, useTheme }}>
         {children}
      </ThemeContext.Provider>
   )
}
