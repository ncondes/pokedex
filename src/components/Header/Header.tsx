import { ThemeButton } from './ThemeButton'
import { Favorites } from './Favorites'
import styles from './Header.module.scss'
import { Home } from './Home'

export const Header = () => {
   return (
      <header className={styles.header}>
         <ul>
            <li>
               <Home />
            </li>
            <div className={styles.theme_and_favorites_links}>
               <li>
                  <ThemeButton />
               </li>
               <li>
                  <Favorites />
               </li>
            </div>
         </ul>
      </header>
   )
}
