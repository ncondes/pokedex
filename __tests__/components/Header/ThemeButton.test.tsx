import { ThemeButton } from '@/components/Header/ThemeButton'
import { ThemeContext } from '@/context/theme'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

describe('ThemeButton', () => {
   it('should render the theme button', () => {
      const { container } = render(
         <ThemeContext.Provider
            value={{
               theme: 'light',
               toggleTheme: jest.fn(),
               useTheme: jest.fn(),
            }}
         >
            <ThemeButton />
         </ThemeContext.Provider>
      )

      const button = container.querySelector('button')!
      const icon = container.querySelector('i')

      expect(button).toBeInTheDocument()
      expect(icon).toHaveClass('fa-regular fa-moon fa-xl')
   })

   it('should match the snapshot', () => {
      const { container } = render(
         <ThemeContext.Provider
            value={{
               theme: 'light',
               toggleTheme: jest.fn(),
               useTheme: jest.fn(),
            }}
         >
            <ThemeButton />
         </ThemeContext.Provider>
      )

      expect(container).toMatchSnapshot()
   })
})
