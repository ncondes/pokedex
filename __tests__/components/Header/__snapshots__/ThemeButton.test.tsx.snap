// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeButton should match the snapshot 1`] = `
<div>
  <button>
    <i
      class="fa-regular fa-moon fa-xl"
    />
  </button>
</div>
`;
